src/Main.elm

```diff
@@ -1,10 +1,177 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
+import Basics.Extra exposing (flip)
+import Browser
+import Data.Audience exposing (Audience)
 import Html exposing (Html)
+import Html.Attributes exposing (style)
+import Html.Events as Events
+import FolderInfo exposing (FolderInfo(..))
+import Maybe.Extra
+import String.Extra
+import Tree.Zipper as Zipper exposing (Zipper)
+import Zipper exposing (kids, tree)
 
 
-main : Html msg
+{- NOTE:    So I am not the most proud of the following code.
+            The styling is pretty arbitrary.
+            The html structure is pretty "medium rare" if you will...
+            In the real world app I think there would be some "company guidelines" or even some code to handle UI - I suppose.
+            So I kinda just went with it and had some fun.
+            The UI is kinda terrible, but I am assuming it may not be the main point of this challenge? Maybe I am wrong.
+            It has been some time since I have gotten my hands this dirty with actual component structures and styling.
+            I feel like it is just a matter of reading couple of articles about what is idiomatic nowadays or something like that.
+            And also - seeing the rest of the project - I feel like one can really infer a lot from some existing code and then just go and replicate that everywhere.
+            Hopefully y'all agree.
+-}
+
+
+
+type Msg
+    = Navigate (Zipper FolderInfo)
+
+
+
+-- I went with the Zipper alone - simply because it is convenient.
+type alias Model =
+    Zipper.Zipper FolderInfo
+
+
+init : Model
+init =
+    tree
+
+
+main : Program () Model Msg
 main =
-    Html.text "There will be app soon!"
+    Browser.sandbox { init = init, update = update, view = view }
+
+
+view : Model -> Html Msg
+view zipper =
+    let
+        (Info { audiences, name, id }) =
+            Zipper.label zipper
+
+        folders =
+            kids zipper
+
+        {- NOTE: I would not really write it like this in real project. Just thought I would have some fun and perhaps humor someone with a pointfree eye ;) -}
+        ifNotRoot =
+            (flip Maybe.andThen <| id) << always << Just
```

Comment:
❤️ 
I like pointfree, I like it that much, so we established some "Are you fcking nuts with those pointfree?" rule, because I was probably only one who always prefer pointfree. But to be fair, it's valid approach do not use it too much, since it make code more difficult to read for new comers and juniors 🤷🏼

Reply:
I agree though.